types/puppeteer/index.d.ts

```diff
@@ -681,7 +681,7 @@ export interface FrameBase {
    */
   $$eval(
     selector: string,
-    fn: (elements: ElementHandle[], ...args: any[]) => any,
+    fn: (elements: NodeListOf<Element>, ...args: any[]) => any,
```

Comment:
The rest arguments can be a `ElementHandle` or anything serializable, thus `ElementHandle|any`.

Last thing; the return value is always of type `Promise<any>`, as it is always a Promise and we don't know the context type upfront.

So `fn`, which should be called `pageFuncton` by the documentation, should be typed like this:

`pageFunction: (elements: NodeListOf<Element>, ...args: (ElementHandle|any)[]) => Promise<any>`

Reply:
Sorry, but I thought `ElementHandle|any` is the same as `any`, is it?